generate_endf71.py

```diff
@@ -99,122 +100,200 @@ def sort_key(path):
+            'compressed_file_size': 9,
+            'uncompressed_file_size': 45
+        },
+        'neutron_wmp': {
+            'base_url': 'https://github.com/mit-crpg/WMP_Library/releases/download/v1.1/',
+            'compressed_files': ['WMP_Library_v1.1.tar.gz'],
+            'file_type': 'wmp',
+            'compressed_file_size': 12,
+            'uncompressed_file_size': 17
+        }
+    }
+}
+
+compressed_file_size, uncompressed_file_size = 0, 0
+for r in release:
+    for p in ('neutron', 'photon', 'neutron_wmp'):
+        compressed_file_size += release_details[release][p]['compressed_file_size']
+        uncompressed_file_size += release_details[release][p]['uncompressed_file_size']
+
+download_warning = """
+WARNING: This script will download up to {} MB of data. Extracting and
+processing the data may require as much as {} MB of additional free disk
+space. This script downloads ENDF/B-VII.1 incident neutron ACE data and
+incident photon ENDF data from NNDC and convert it to an HDF5 library
+for use with OpenMC.
+""".format(compressed_file_size, uncompressed_file_size)
+
+
+# ==============================================================================
+# DOWNLOAD FILES FROM NNDC SITE
+
+if args.download:
+    print(download_warning)
+    for particle in args.particles:
+        details = release_details[release][particle]
+        for i, f in enumerate(details['compressed_files']):
+            url = details['base_url'] + f
+            if 'checksums' in details.keys():
+                checksum = details['checksums'][i]
+                downloaded_file = download(url,
+                                           output_path=download_path / particle,
+                                           checksum=checksum)
+            else:
+                downloaded_file = download(url,
+                                           output_path=download_path / particle,
+                                           )
+
+# ==============================================================================
+# EXTRACT FILES FROM TGZ
+
+if args.extract:
+    for particle in args.particles:
+
+        if release_details[release][particle]['file_type'] == 'wmp':
+            extraction_dir = wmp_files_dir
+        elif release_details[release][particle]['file_type'] == 'endf':
+            extraction_dir = endf_files_dir
+
+        for f in release_details[release][particle]['compressed_files']:
+
+            # Extract files different depending on compression method
+            if f.endswith('.zip'):
+                with zipfile.ZipFile(download_path / particle / f, 'r') as zipf:
+                    print(f'Extracting {f}...')
+                    zipf.extractall(extraction_dir)
+            else:
+                with tarfile.open(download_path / particle / f, 'r') as tgz:
+                    print(f'Extracting {f}...')
+                    tgz.extractall(path=extraction_dir)
```

Comment:
Ok, so I might have changed my mind on this. I didn't realize that the way things are set up now, the `WMP_Library` directory will actually appear in the final library, which is undesirable. Given that, I'd say go ahead and go back to approach you had before -- extract the members one-by-one and remove the leading directory.

Reply:
Now extracting tgz without the internal folder structure to avoid dir ending up in the cross_sections.xml file